Sources/BraveShields/WebcompatReporter.swift

```diff
@@ -8,77 +8,180 @@ import Shared
 
     var components = URLComponents()
     components.scheme = "https"
-    components.host = baseURL
+    components.host = baseHost
     components.path = "/\(version)/webcompat"
 
-    guard let baseDomain = url.baseDomain,
-      let key = apiKey,
-      let endpoint = components.url
-    else {
-      Logger.module.error("Failed to setup webcompat request")
+    guard let endpoint = components.url else {
+      Self.log.error("Failed to setup webcompat request")
       return false
     }
 
-    // We want to ensure that the URL _can_ be normalized, since `domainURL` will return itself
-    // (the full URL) if the URL can't be normalized. If the URL can't be normalized, send only
-    // the base domain without scheme.
-    let domain = url.normalizedHost() != nil ? url.domainURL.absoluteString : baseDomain
-
-    let payload = [
-      "domain": domain,
-      "api_key": key,
-    ]
-
     do {
+      let encoder = JSONEncoder()
       var request = URLRequest(url: endpoint)
       request.httpMethod = "POST"
-      request.httpBody = try JSONSerialization.data(withJSONObject: payload, options: [])
+      request.httpBody = try encoder.encode(payload)
+      
       if let userAgent = userAgent {
         request.setValue(userAgent, forHTTPHeaderField: "User-Agent")
       }
-
+      
       let session = URLSession(configuration: .ephemeral)
-      return await withCheckedContinuation { continuation in
-        let task = session.dataTask(with: request) { data, response, error in
-          var success: Bool = true
-          if let error = error {
-            Logger.module.error("Failed to report webcompat issue: \(error.localizedDescription)")
-            success = false
-          }
-          if let response = response as? HTTPURLResponse {
-            success = response.statusCode >= 200 && response.statusCode < 300
-            if !success {
-              Logger.module.error("Failed to report webcompat issue: Status Code \(response.statusCode)")
-            }
-          }
-          continuation.resume(returning: success)
+      let result = try await session.data(for: request)
+      
+      if let response = result.1 as? HTTPURLResponse {
+        let success = response.statusCode >= 200 && response.statusCode < 300
+        
+        if !success {
+          log.error("Failed to report webcompat issue: Status Code \(response.statusCode)")
         }
-        task.resume()
+        
+        return success
+      } else {
+        return false
       }
     } catch {
       Logger.module.error("Failed to setup webcompat request payload: \(error.localizedDescription)")
       return false
     }
   }
 }
+
+private extension ShieldLevel {
+  /// The value that is sent to the webcompat report server
+  var reportLabel: String {
+    switch self {
+    case .aggressive: "aggressive"
+    case .standard: "standard"
+    case .disabled: "allow"
+    }
```

Comment:
CI is giving some unreasonable build error here

Reply:
Missing returns maybe

Reply:
This is valid in Xcode 15/Swift 5.9, but yeah for now you will need returns until we shift over